### PR TITLE
Us167  As a developer, I want to prototype UI panel that shows an overview of current player turn, dice roll, and money so that players can see key info during gameplay.

### PR DESCRIPTION
**Pull request by kescob0825**
Acceptance Criteria:
- Turn UI shows player name and color/game piece
- Dice results displayed after roll
- Money Display updates on turn end
- All UI data driven from events including TurnStarted, DiceRolled, PlayerMoneyChanged

Tasks:
- 168 Create UI as basic panel elements
- 169 Bind UI controlled to event channels
- 170 Connect data update updates to text elements in UI
- 171 Create a Play test for layout clarity
- 216 DEPENDENCY ON DICEMANAGER
- 230 Create an event channel to request the DiceManager to roll the dice.

### CODE REVIEW

**Comment by hdathert** [comment]
Went ahead and resolved the merge conflict on SampleScene to accept this version over what's in `dev` currently.

File: Assets/Scripts/Events/EventChannelTypes/DiceRolledEventChannel.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ﻿using UnityEngine;
+using static UnityEngine.UI.CanvasScaler;
```
Comment (hdathert) [inline Assets/Scripts/Events/EventChannelTypes/DiceRolledEventChannel.cs]: Why does this need to be here?

File: Assets/Scripts/Managers/DiceManager.cs [hunk 2]
```diff
@@ -20,7 +20,27 @@ public class DiceManager : MonoBehaviour
...
+                this);
+            return;
+        }
+
+        rollDiceRequestedChannel.Subscribe(RollDiceRequest);
+    }
+    private void OnDisable()
+    {
+        if (rollDiceRequestedChannel != null)
+            rollDiceRequestedChannel.Unsubscribe(RollDiceRequest);
```
Comment (hdathert) [inline Assets/Scripts/Managers/DiceManager.cs]: This totally works but can be 'cleaned up' to use the `?` syntax:
`rollDiceRequestedChannel?.Unsubscribe(RollDiceRequest);`

File: Assets/Tests/Shared/ManagerTestBase.cs [hunk 3]
```diff
@@ -16,7 +16,12 @@ protected T CreateChannel<T>() where T : ScriptableObject
             eventChannels.Add(channel);
             return channel;
         }
-    
+        protected T CreateComponent<T>(string componentName, GameObject root) where T : Component
```
Comment (hdathert) [inline Assets/Tests/Shared/ManagerTestBase.cs]: This seems misnamed- `AddComponent<T>` is already a method of `GameObject`. What this is actually doing is creating a child `GameObject` and then attaching a `Component` to it.

**Review by hdathert — CHANGES_REQUESTED** [review]
# Code Review Checklist

**Completed for**: US #167 and Tasks #168 to #230  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

---

Just a couple of requested changes here, but mostly looks solid. Its hard to tell if its all working or not without a GameManager in the actual scene orchestrating the whole thing.

File: Assets/Scripts/Events/EventChannelTypes/DiceRolledEventChannel.cs (hunk 1 above)
Comment (kescob0825) [reply Assets/Scripts/Events/EventChannelTypes/DiceRolledEventChannel.cs]: Went ahead and deleted this.

File: Assets/Scripts/Managers/DiceManager.cs (hunk 2 above)
Comment (kescob0825) [reply Assets/Scripts/Managers/DiceManager.cs]: This is now cleaned up

File: Assets/Tests/Shared/ManagerTestBase.cs (hunk 3 above)
Comment (kescob0825) [reply Assets/Tests/Shared/ManagerTestBase.cs]: Changed the naming from CreateComponent to CreateAndAttachComponent.

**Review by dzadroga — APPROVED** [review]
# Code Review Checklist

**Completed for**: US 167 and Tasks 168 to 171, Task 216 and Task 230
**Code Review Completed By**: Dakota Zadroga

Use this checklist to review new code changes before merging.
---
## Implementation
- [x ] Does this code change accomplish what it is supposed to do?
- [x ] Can the solution be simplified?
- [x ] This change does **not** add unwanted compile-time or run-time dependencies.
- [ x] The code is at the right abstraction level.
- [ x] The code is modular enough.
- [ x] The solution is adequate for maintainability, readability, performance, and security.
- [x ] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x ] No additional best practices or design patterns would substantially improve this code.
---
## Logic Errors and Bugs
- [x ] No use cases where the code behaves incorrectly.
- [x ] No inputs or external events that can break the code.
- [x ] Code agrees with specifications.
- [x ] No off-by-one errors.
- [x ] No magic numbers (or they are well-named constants).
- [ x] Proper use of global/large-scope variables (minimized and justified).
---
## Error Handling and Logging
- [ x] Error handling is implemented correctly.
- [ x] Error messages are user-friendly (where applicable).
- [x ] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).
---
## Dependencies
- [x ] Documentation/configuration/README updates were made as needed.
- [x ] No negative impact on other parts of the system or backward compatibility.
---
## Security and Data Privacy
- [ x] No new security vulnerabilities introduced.
- [ x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [ x] Data from external APIs/libraries is handled safely and checked as needed.
---
## Performance
- [x ] No significant performance regressions.
- [ x] No obvious opportunities for significant performance gains left unaddressed.
---
## Ethics and Morality
- [x ] Appropriate measures exist to prevent or report harassment/abuse (if applicable).
---
## Testing and Testability
- [x ] Code is testable.
- [ x] Automated tests have been added or updated.
- [ x] Existing tests sufficiently cover the change (unit/integration/system).
- [ x] Edge cases and boundary inputs are tested.
---
## Readability
- [ x] Code is easy to understand.
- [ x] Methods/functions are appropriately small and focused.
- [ x] Names (functions/classes/variables) are clear and descriptive.
- [ x] Code is in the correct file/folder/package.
- [ x] Control flow is intuitive; data flow is clear.
- [ x] Comments are clear, relevant, and up to date.
- [ x] No commented-out or redundant code remains.
- [ x] No overly dense one-liners; no overly long methods.
---
**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.
*Based on awesome code review guides and MIT’s code review recommendations.*

**Review by hdathert — APPROVED** [review]

**Comment by hdathert** [comment]
# Code Review Checklist

**Completed for**: US #167 and Tasks #168 to #230  
**Code Review Completed By**: @hdathert 

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*

---
